qiskit/visualization/pulse_v2/drawing_objects.py

```diff
@@ -0,0 +1,251 @@
+
+import numpy as np
+
+from qiskit import pulse
+
+
+class ElementaryData(ABC):
+    """Abstract class of visualization intermediate representation."""
+    def __init__(self,
+                 data_type: str,
+                 channel: pulse.channels.Channel,
+                 meta: Union[Dict[str, Any], None],
+                 offset: float,
+                 visible: bool,
+                 styles: Dict[str, Any]):
+        """Create new visualization IR.
+        Args:
+            data_type: String representation of this drawing object.
+            channel: Pulse channel object bound to this drawing.
+            meta: Meta data dictionary of the object.
+            offset: Offset coordinate of vertical axis.
+            visible: Set ``True`` to show the component on the canvas.
+            styles: Style keyword args of the object. This conforms to `matplotlib`.
+        """
+        self.data_type = data_type
+        self.channel = channel
+        self.meta = meta
+        self.offset = offset
+        self.visible = visible
+        self.styles = styles
+
+    @property
+    @abstractmethod
+    def data_key(self):
+        """Return unique hash of this object."""
+        pass
+
+    def __repr__(self):
+        return "{}(type={}, key={})".format(self.__class__.__name__,
+                                            self.data_type,
+                                            self.data_key)
+
+    def __eq__(self, other):
+        return isinstance(other, self.__class__) and self.data_key == other.data_key
+
+
+class FilledAreaData(ElementaryData):
+    """Drawing IR to represent object appears as a filled area.
+    This is the counterpart of `matplotlib.axes.Axes.fill_between`.
+    """
+    # pylint: disable=invalid-name
+    def __init__(self,
+                 data_type: str,
+                 channel: pulse.channels.Channel,
+                 x: np.ndarray,
+                 y1: np.ndarray,
+                 y2: np.ndarray,
+                 meta: Union[Dict[str, Any], None],
+                 offset: float,
+                 visible: bool,
+                 styles: Dict[str, Any]):
+        """Create new visualization IR.
+        Args:
+            data_type: String representation of this drawing object.
+            channel: Pulse channel object bound to this drawing.
+            x: Series of horizontal coordinate that the object is drawn.
+            y1: Series of vertical coordinate of upper boundary of filling area.
+            y2: Series of vertical coordinate of lower boundary of filling area.
+            meta: Meta data dictionary of the object.
+            offset: Offset coordinate of vertical axis.
+            visible: Set ``True`` to show the component on the canvas.
+            styles: Style keyword args of the object. This conforms to `matplotlib`.
+        """
+        self.x = x
+        self.y1 = y1
+        self.y2 = y2
+
+        super().__init__(
+            data_type=data_type,
```

Comment:
Should standardize on this syntax or the syntax in `def __init__`. Personally I prefer this syntax, but others prefer the above.